### PR TITLE
test: Add missing tests for recent fixes and features

### PR DESCRIPTION
**Pull request by dav-pascual**
Specifics in commit

### CODE REVIEW

File: src/mrack/providers/provider.py [hunk 1]
```diff
@@ -432,7 +436,7 @@ async def strategy_retry(self, reqs):
         success_hosts = []
         error_hosts = []
         delta_sleep = global_context.CONFIG.delta_sleep
-        max_utilization = global_context.CONFIG.max_utilization
+        max_utilization = self._get_max_utilization()
```
Comment (Tiboris) [inline src/mrack/providers/provider.py:439]: One nitpick, could you please extract this to another 'refactor' commit please?

**Review by Tiboris — CHANGES_REQUESTED** [review]
One minor nitpick David, and I think we can go for merge. Do you agree?

File: src/mrack/providers/provider.py (hunk 1 above)
Comment (dav-pascual) [reply src/mrack/providers/provider.py:439]: No problem, updated

**Review by Tiboris — APPROVED** [review]
Thanks for change!

**Pull request merged**